### ISSUE TITLE
Why do you flip the xyz here?

### ISSUE THREAD

**Issue by Chinartist** [LinShan-Bin/OccNeRF#12]
` ind_norm_sem = ind_norm_sem.flip((-1,))`
In my opinion, the xyz has been converted to the world coordiante. So if flip the xyz,  the function will sample grid along the wrong axis（x along z-axis, z along x-axis）. 
Is there anything wrong with my idea？

**Comment by Lazyangel** [comment]
Hello, have you found the answer to this question yet? Is the flipping operation here related to coordinate system transformation?

**Comment by Chinartist** [comment]
@Lazyangel ，it is because the F.gridsample function will sample the x along the last dim, sample the y along the mid dim, sample the z along first dim,  so you need to flip the [x,y,z] to [z,y,x] so that you can sample the correct feature

**Comment by Lazyangel** [comment]
Thank you for such a quick reply!

**Comment by Lazyangel** [comment]
> @Lazyangel ，it is because the F.gridsample function will sample the x along the last dim, sample the y along the mid dim, sample the z along first dim, so you need to flip the [x,y,z] to [z,y,x] so that you can sample the correct feature

Hello, I have a question. Why there is no need to flip when sampling from image features?
```
def unproject_image_to_mem(self, rgb_camB, pixB_T_camA, camB_T_camA, Z, Y, X, assert_cube=False):
        """2D to 3D lifting

        Args:
            rgb_camB (torch.Tensor): Image features. [B, C, H // 4, W // 4] (6, 64, 84, 168)
            pixB_T_camA (torch.Tensor): Projection matrix. From world coordinates to pixel coordinates. [B, 4, 4] (6, 4, 4) 
            camB_T_camA (torch.Tensor): Projection matrix. From world coordinates to camera coordinates. [B, 4, 4] (6, 4, 4) 
            Z, Y, X (int): Voxel size.

        Returns:
            torch.Tensor: Voxel features. [B, C, Z, Y, X] (6, 64, 16, 256, 256)
        """
        B, C, H, W = list(rgb_camB.shape)
        xyz_memA = basic.gridcloud3d(B, Z, Y, X, norm=False, device=pixB_T_camA.device) 
        xyz_camA = self.Mem2Ref(xyz_memA, Z, Y, X, assert_cube=assert_cube)  # xyz in world coordinates

        xyz_camB = geom.apply_4x4(camB_T_camA, xyz_camA)  # xyz in camera coordinates

        z = xyz_camB[:,:,2]  

        xyz_pixB = geom.apply_4x4(pixB_T_camA, xyz_camA)
        normalizer = torch.unsqueeze(xyz_pixB[:,:,2], 2) # xyz_pixB[:,:,2] = [B, N] ——> [B, N, 1]
        EPS=1e-6
        # z = xyz_pixB[:,:,2] 
        xy_pixB = xyz_pixB[:,:,:2]/torch.clamp(normalizer, min=EPS)  # xyz in pixel coordinates
        # this is B x N x 2
        # this is the (floating point) pixel coordinate of each voxel
        x, y = xy_pixB[:,:,0], xy_pixB[:,:,1]
        # these are B x N
        x_valid = (x>-0.5).bool() & (x<float(W-0.5)).bool()
        y_valid = (y>-0.5).bool() & (y<float(H-0.5)).bool()
        z_valid = (z>0.0).bool()
        valid_mem = (x_valid & y_valid & z_valid).reshape(B, 1, Z, Y, X).float()


        # native pytorch version
        y_pixB, x_pixB = basic.normalize_grid2d(y, x, H, W)  # normalize to [-1, 1] 
        # since we want a 3d output, we need 5d tensors
        z_pixB = torch.zeros_like(x)
        xyz_pixB = torch.stack([x_pixB, y_pixB, z_pixB], axis=2) 

        rgb_camB = rgb_camB.unsqueeze(2) # [B, C, 1， H // 4, W // 4]
        xyz_pixB = torch.reshape(xyz_pixB, [B, Z, Y, X, 3])
        # grid_sample(input, grid) 
        # Given an input and a flow-field grid, computes the output using input values and pixel locations from grid.
        values = F.grid_sample(rgb_camB, xyz_pixB, align_corners=False) # (N, H, W, Z, 3)

        values = torch.reshape(values, (B, C, Z, Y, X))
        # 24, 300, 300
        values = values * valid_mem

        return values
```
